dev_requirements.txt

```diff
@@ -0,0 +1,5 @@
+black

```

Comment:
Have you considered using poetry or rye to manage dependencies? As you also introduced ruff (which I like!), we might also consider having a nice tool for the dependencies in place. :)

Reply:
I've never worked with poetry till now. What about pip-tools, if it is only depencecies?

Reply:
I'm also fine to keep it as it is right now as its not too complex.